src/bun.js/node/types.zig

```diff
@@ -1840,548 +1915,2971 @@ pub const Emitter = struct {
+                    // overwritten by "\\\\?\\UNC\\" which is 8 bytes long.
+                    // Use bun.copy because resolvedPath and buf overlap.
+                    bun.copy(T, buf[bufOffset ..bufSize], resolvedPath);
+                    // Equiv to std.os.windows.NamespacePrefix.verbatim
+                    // https://github.com/ziglang/zig/blob/dcaf43674e35372e1d28ab12c4c4ff9af9f3d646/lib/std/os/windows.zig#L2358-L2374
+                    buf[0] = CHAR_BACKWARD_SLASH;
+                    buf[1] = CHAR_BACKWARD_SLASH;
+                    buf[2] = CHAR_QUESTION_MARK;
+                    buf[3] = CHAR_BACKWARD_SLASH;
+                    buf[4] = 'U';
+                    buf[5] = 'N';
+                    buf[6] = 'C';
+                    buf[7] = CHAR_BACKWARD_SLASH;
+                    return MaybeSlice(T) { .result = buf[0 ..bufSize] };
+                }
+            }
+        } else if (
+            isWindowsDeviceRootT(T, byte0) and
+            resolvedPath[1] == CHAR_COLON and
+            resolvedPath[2] == CHAR_BACKWARD_SLASH
+        ) {
+            // Matched device root, convert the path to a long UNC path
+
+            // Translated from the following JS code:
+            //   return `\\\\?\\${resolvedPath}`
+            bufOffset = 4;
+            bufSize = len + 4;
+            // Move all bytes to the right by 4
+            // Use bun.copy because resolvedPath and buf overlap.
+            bun.copy(T, buf[bufOffset ..bufSize], resolvedPath);
+            // Equiv to std.os.windows.NamespacePrefix.verbatim
+            // https://github.com/ziglang/zig/blob/dcaf43674e35372e1d28ab12c4c4ff9af9f3d646/lib/std/os/windows.zig#L2358-L2374
+            buf[0] = CHAR_BACKWARD_SLASH;
+            buf[1] = CHAR_BACKWARD_SLASH;
+            buf[2] = CHAR_QUESTION_MARK;
+            buf[3] = CHAR_BACKWARD_SLASH;
+            return MaybeSlice(T) { .result = buf[0 ..bufSize] };
+        }
+        return MaybeSlice(T) { .result = path };
+    }
+
+    pub inline fn toNamespacedPathWindows(path: []const u8, buf: []u8, varedLenBuf: []u8) MaybeSlice(u8) {
+        return toNamespacedPathWindowsT(u8, path, buf, varedLenBuf);
+    }
+
+    pub fn toNamespacedPath(globalObject: *JSC.JSGlobalObject, isWindows: bool, args_ptr: [*]JSC.JSValue, args_len: u16) callconv(.C) JSC.JSValue {
+        if (comptime is_bindgen) return JSC.JSValue.jsUndefined();
+        if (args_len == 0) return JSC.JSValue.jsUndefined();
+        var path_ptr = args_ptr[0];
+        // Act as an identity function for non-string values and non-Windows platforms.
+        // Note: this will *probably* throw somewhere.
+        if (!isWindows or !path_ptr.isString()) return path_ptr;
+
+        var stack_fallback = std.heap.stackFallback(
+            stack_fallback_size_small, 
+            JSC.getAllocator(globalObject)
+        );
+        const allocator = stack_fallback.get();
+
+        const pathZSlice = path_ptr.toSlice(globalObject, allocator);
```

Comment:
☝️ The only place we jump straight to `pathZSlice` instead of going to `pathZStr`.